lib/ansible/galaxy/api.py

```diff
@@ -308,6 +309,9 @@ def __init__(
         if not no_cache:
             self._cache = _load_cache(self._b_cache_path)
 
+        # For efficiency, this metadata is populated when listing collection versions.
+        self.requires_ansible = {}
```

Comment:
Have you considered using a `defaultdict` here?
I think that it'll simplify entry creation on access below.
```suggestion
        self.requires_ansible = collections.defaultdict(dict)
```